### ISSUE TITLE
An attendee is still an active speaker when he stops to make any sound or mutes himself

### ISSUE THREAD

**Issue by rlebosse** [aws/amazon-chime-sdk-js#1132]
### What are you trying to do?

* I'm monitoring the active speakers with the following code :

```js
const onActiveSpeakers = activeSpeakers => {
    console.log('rlebosse activeSpeakers', activeSpeakers)
}
audioVideo.subscribeToActiveSpeakerDetector(
    new DefaultActiveSpeakerPolicy(),
    onActiveSpeakers,
    scores => {
        console.log('rlebosse scores', scores)
    },
    1000
)
```

* I use this piece of code in order to keep track of the speaking attendees in order to be able to determine how much time they speak (if they speak a short time : no need to flag them as real speaking attendees).

For a Chime Meeting in which there is only one attendee (USER_1) :
* USER_1 speaks : `onActiveSpeakers` is called with USER_1 attendee id in `activeSpeakers` ✅ 
* USER_1 stops speaking or mutes himself : `onActiveSpeakers` is not called with an empty array for `activeSpeakers` as soon as USER_1 becomes silent ❌
* `onActiveSpeakers` is called with with an empty array for `activeSpeakers` after a long time

This prevents me from knowing who is really speaking in real time and be able to know how much time they spoke.

However when there are 2 attendees : USER_1 and USER_2
* USER_1 speaks : `onActiveSpeakers` is called with USER_1 attendee id in `activeSpeakers` ✅ 
* USER_1 stops speaking or mutes himself
* USER_2 speaks right after USER_1 became silent : `onActiveSpeakers` is called with USER_2 attendee id and without USER_1 in `activeSpeakers` ✅ 

Could you help me on this ?

Thanks a lot and keep on the great work, cheers !

**Platform**
 - OS: macOS
 - Browser: Google Chrome
 - Browser Version: 89
 - Amazon Chime SDK version: 2.4.1

**Comment by ltrung** [comment]
From the code you set the callback interval `scoresCallbackIntervalMs` to 1000ms which seems a bit high. Could you reduce it to 100ms and see if that would fix the issue for you?

**Comment by rlebosse** [comment]
Thanks for your answer @ltrung.

I am not sure to understand what is the real purpose of `scoresCallbackIntervalMs`.

* Does this interval represents at which frequency the attendee scores will be re-calculated ?
or
* Does this interval represents at which frequency the following callback will be called ?

```js
scores => {
        console.log('rlebosse scores', scores)
    }
```

If it the 2nd option, it has no impact on the fact that a user will stay "active speaker" whereas he stopped making any sound or muted himself ?

Thanks !

**Comment by ltrung** [comment]
@rlebosse Sorry I misread the question. The behavior you described is controlled by the default active speaker policy. More specifically the score is calculated using [speaker weight](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/activespeakerpolicy/DefaultActiveSpeakerPolicy.ts#L18). The default policy is set to 0.9 so it prioritizes the existing score more than the new volume value and thus, it would take a while for the score to get to 0 and the active speaker would remain to be the same when there is only one attendee as you pointed out. 
When there is more than one attendee and the other attendee speaks, the previous active speaker will be taken over (this is affected by [take over rate](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/activespeakerpolicy/DefaultActiveSpeakerPolicy.ts#L36)) and thus, the active speaker will be updated immediately.
In summary, this policy tries to mimic the behavior in a meeting where we will keep our head toward the current active speaker even though he might stop speaking and only turn toward another if they speak.

For your use case, could you try to tweak the default active speaker policy and reduce the speaker weight. Alternatively, you can try to use [volume indicator](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/audiovideofacade/DefaultAudioVideoFacade.ts#L243)

**Comment by rlebosse** [comment]
Thanks for all these precisions @ltrung, i'll try using volume indicator !
